standard/expressions.md

```diff
@@ -1470,6 +1477,23 @@ parenthesized_expression
 
 A *parenthesized_expression* is evaluated by evaluating the *expression* within the parentheses. If the *expression* within the parentheses denotes a namespace or type, a compile-time error occurs. Otherwise, the result of the *parenthesized_expression* is the result of the evaluation of the contained *expression*.
 
+### §tuple-expressions-new-clause Tuple expressions
+
+A *tuple_expression* consists of two or more comma-separated and optionally-named *expression*s enclosed in parentheses.
+
+``` ANTLR
+tuple_expression
+    : `(` tuple_element (',' tuple_element)+ ')'
+tuple_element
+    : (identifier ':')? expression
+```
+
+A tuple expression is classified as a tuple. It has a type if all the element expressions have a type. In that case, the type of a tuple expression is the tuple type of the same arity, where each type element has the type of the corresponding element expression and the same identifier as the corresponding tuple element if it has one, or none if it does not.
```

Comment:
grammar concern:  I find this sentence hard to parse correctly. Does "or none if it does not" apply to the names of tuple elements, the type of each element, or the tuple itself?

Reply:
Reworded in upcoming commit to "... where each type element has the type - and name, where given - of the corresponding element expression."